filebeat/module/system/syslog/test/darwin-syslog-sample.log-expected.json

```diff
@@ -3,27 +3,27 @@
-        "system.syslog.pid": "21412", 
-        "system.syslog.program": "GoogleSoftwareUpdateAgent", 
+        "message": "2016-12-13 11:35:28.420 GoogleSoftwareUpdateAgent[21412/0x700007399000] [lvl=2] -[KSAgentApp updateProductWithProductID:usingEngine:] Checking for updates for \"All Products\" using engine <KSUpdateEngine:0x100341a00\n\t\tticketStore=<KSPersistentTicketStore:0x100204520 store=<KSKeyedPersistentStore:0x100213290\n\t\t\tpath=\"/Users/tsg/Library/Google/GoogleSoftwareUpdate/TicketStore/Keystone.ticketstore\"\n\t\t\tlockFile=<KSLockFile:0x1002160e0\n\t\t\t\tpath=\"/Users/tsg/Library/Google/GoogleSoftwareUpdate/TicketStore/Keystone.ticketstore.lock\"\n\t\t\t\tlocked=NO\n\t\t\t>\n\t\t>>\n\t\tprocessor=<KSActionProcessor:0x1003bb5f0\n\t\t\tdelegate=<KSUpdateEngine:0x100341a00>\n\t\t\tisProcessing=NO actionsCompleted=0 progress=0.00\n\t\t\terrors=0 currentActionErrors=0\n\t\t\tevents=0 currentActionEvents=0\n\t\t\tactionQueue=( )\n\t\t>\n\t\tdelegate=(null)\n\t\tserverInfoStore=(null)\n\t\terrors=0\n\t>", 
+        "process.name": "GoogleSoftwareUpdateAgent", 
+        "process.pid": "21412", 
         "system.syslog.timestamp": "Dec 13 11:35:28"
     }, 
     {
         "@timestamp": "2018-12-13T11:35:28.000Z", 
         "event.dataset": "syslog", 
         "event.module": "system", 
+        "host.hostname": "a-mac-with-esc-key", 
         "input.type": "log", 
         "log.offset": 907, 
-        "system.syslog.hostname": "a-mac-with-esc-key", 
-        "system.syslog.message": "2016-12-13 11:35:28.421 GoogleSoftwareUpdateAgent[21412/0x700007399000] [lvl=2] -[KSUpdateEngine updateAllExceptProduct:] KSUpdateEngine updating all installed products, except:'com.google.Keystone'.", 
-        "system.syslog.pid": "21412", 
-        "system.syslog.program": "GoogleSoftwareUpdateAgent", 
+        "message": "2016-12-13 11:35:28.421 GoogleSoftwareUpdateAgent[21412/0x700007399000] [lvl=2] -[KSUpdateEngine updateAllExceptProduct:] KSUpdateEngine updating all installed products, except:'com.google.Keystone'.", 
+        "process.name": "GoogleSoftwareUpdateAgent", 
+        "process.pid": "21412", 
```

Comment:
IIRC process.pid needs converted to number.

Reply:
I have the same issue in an other PR. I need to investigate if during the testing the ECS templates are also properly loaded.

Reply:
This PR has not changed the fact that the PID was never properly cast to an int.

It's a trivial thing to fix in the grok, however. I'd be ok with fixing that in places where I see these values improperly cast. Could this introduce a breaking change, though? Visualizations are not affected by the field in `_source` being a string or an int, so that part should be fine. But are there other things that could be affected by these values going from a string to an int?

Reply:
Actually `haproxy.log` has it the worst. It has a mix of int and strings, both for the PID and for the port. If this was a problem, we would have heard about it by now. I'll go around and do this small improvement on all of my open PRs.

Reply:
It's a rabbit hole, though. Now I notice that IIS response and response times are not converted to ints...

Reply:
I don't think it's a grok issue here. It's more of a template issue in the tests.

Reply:
No, all of these fields have proper `long` mappings. It really is as simple as adding `:int` at the end of the GROK expression, so that the value in the JSON document is a integer rather than a quoted string.

I've actually gone deep in the rabbit hole and changed all of the occurrences I saw in this PR and all of the other ones I'm working on at the moment.